pynpoint/util/postproc.py

```diff
@@ -0,0 +1,158 @@
+    mask : np.ndarray
+        Mask (2D).
+    processing_type : str
+        Post-processing type:
+            - ADI: Angular differential imaging.
+            - SDI: Spectral differential imaging.
+            - SDI+ADI: Spectral and angular differential imaging.
+            - ADI+SDI: Angular and spectral differential imaging.
+
+    Returns
+    -------
+    np.ndarray
+        Residuals of the PSF subtraction.
+    np.ndarray
+        Derotated residuals of the PSF subtraction.
+    """
+
+    if not isinstance(pca_number, tuple):
+        pca_number = (pca_number, -1)
+
+    if mask is None:
+        mask = 1.
+
+    res_raw = np.zeros(images.shape)
+    res_rot = np.zeros(images.shape)
+
+    if processing_type == 'ADI':
+        if images.ndim == 2:
+            res_raw, res_rot = pca_psf_subtraction(images=images*mask,
+                                                   angles=angles,
+                                                   scales=None,
+                                                   pca_number=pca_number[0],
+                                                   pca_sklearn=pca_sklearn,
+                                                   im_shape=im_shape,
+                                                   indices=indices)
+
+        elif images.ndim == 4:
+            for i in range(images.shape[0]):
+                res_raw[i, ], res_rot[i, ] = pca_psf_subtraction(images=images[i, ]*mask,
+                                                                 angles=angles,
+                                                                 scales=None,
+                                                                 pca_number=pca_number[0],
+                                                                 pca_sklearn=pca_sklearn,
+                                                                 im_shape=im_shape,
+                                                                 indices=indices)
+
+    elif processing_type == 'SDI':
+        for i in range(images.shape[1]):
+            im_scaled = sdi_scaling(images[:, i, :, :], scales)
+
+            res_raw[:, i], res_rot[:, i] = pca_psf_subtraction(images=im_scaled*mask,
+                                                               angles=np.full(scales.size,
+                                                                              angles[i]),
+                                                               scales=scales,
+                                                               pca_number=pca_number[0],
+                                                               pca_sklearn=pca_sklearn,
+                                                               im_shape=im_shape,
+                                                               indices=indices)
+
+    elif processing_type == 'SDI+ADI':
+        # SDI
+        res_raw_int = np.zeros(res_raw.shape)
+
+        for i, _ in enumerate(images[0]):
```

Comment:
Change this to something easier to undertsnad? E.g. `for i in range(image.shape[1])`. Also in the other for loops below. Not sure if my suggestion is correct though.

Reply:
I changed it in the suggested way.